rocketpy/Rocket.py

```diff
@@ -209,14 +209,12 @@ def __init__(
         # Parachute data initialization
         self.parachutes = []
 
-        # Rail button data initialization
-        self.railButtons = None
-
         # Aerodynamic data initialization
         self.aerodynamicSurfaces = AeroSurfaces()
         self.nosecone = AeroSurfaces()
         self.fins = AeroSurfaces()
         self.tail = AeroSurfaces()
+        self.rail_buttons = AeroSurfaces()
```

Comment:
Sorry for the question, but why, again, is everything here AeroSurfaces?

From what I understand, #341 modified the AeroSurfaces class to make the position argument optional... which as far as I know literally makes the AeroSurfaces class degenerate back to the list class when no positional argument is used.  So I am not sure it really make sense to me, in terms of code architecture, to have position argument as optional. When we don't need it, we can just use a list, instead of the AeroSurfaces class.

Furthermore, how are rail_buttons related to aerodynamic surfaces?

Reply:
The AeroSurface class has been changed to act as a list, if there is more then one item in `AeroSurfaces._aerosurfaces`, and act as just the first item of `AeroSurfaces._aerosurfaces` if not.

Example:
If you create a rocket and then add a nose cone to it, you can access the NoseCone object in a few ways:

- With `Rocket.aerodynamicSurface[0]`
- Or `Rocket.nosecone`

If you then add another nosecone and a finset, the  `Rocket.aerodynamicSurface` will be a list with both NoseCones and Fins, while `Rocket.nosecone` will be a list with both NoseCone objects. So to access the first added nosecone now, you can do:

- With `Rocket.aerodynamicSurface[0]`
- Or `Rocket.nosecone[0]`

Essentially what the `AeroSurfaces class` is doing is acting as a list when it needs to, and acting as just an object when a list is not needed.

All of this makes accessing some information in the rocket class much easier and it really makes things easier in the Dispersion Class

> From what I understand, #341 modified the AeroSurfaces class to make the position argument optional

We use the position argument only when appending to the `Rocket.aerodynamicSurfaces`, because it is the only time where it is useful

Reply:
I am having quite a hard timing trying to understand all the different ways that `positions` are being handled:

- Why does all of the aerosurfaces (like `Nosecone`, `Fins` etc) have an attribute `self.position = None` that is not documented? Who sets or uses that value?
- What is the difference between this parameter and the `position` parameter that is appended to the `AeroSurfaces` list? If they are not that same, they must not have the same name.